src/cdomains/intDomain.ml

```diff
@@ -128,6 +138,11 @@ sig
   val of_interval: Cil.ikind -> int_t * int_t -> t
   val is_top_of: Cil.ikind -> t -> bool
   val invariant_ikind : Invariant.context -> Cil.ikind -> t -> Invariant.t
+
+  val refine_with_congruence: Cil.ikind -> t -> (int_t * int_t) option -> t
+  val refine_with_interval: Cil.ikind -> t -> (int_t * int_t) option -> t
```

Comment:
Can we use `Interval.t` and `Congruences.t` here instead of `(int_t * int_t) option`? Or does that lead to issues with cyclic builds?

Reply:
I wanted to do that at first. This is in the definition of type S. At this point the module Congruences is not yet defined. But we can't put its definition before that as Congruences has type S. Same situation with Interval.